### DISCUSSION TITLE
Happy Quarkus-CXF Discussions

### DISCUSSION THREAD

**Discussion by wh81752** [quarkiverse/quarkus-cxf#229] [General]
I'm glad to see a discussion forum here.

**Comment by fiodorov70** [comment]
Is client certificate authentication supported  by Quarkus-CXF?
Thanks.

**Reply by dufoli** [reply]
Can you précise the kind of certificate? I have never test but it must work...

**Reply by fiodorov70** [reply]
I mean p12 certificate: we shoud call a web service that requires Client Authentication. Thanks.

**Reply by dufoli** [reply]
In native?
In jvm? 
, have you got a link to doc? 
Jvm it must work only issue can be to configure client but @shumonsharif have provide few trick with configurer function

**Reply by fiodorov70** [reply]
I would like in native, if possible.
Now we set the client certificate in AbstractPhaseInterceptor.handleMessage handler, calling tlsClientParameters.setSSLSocketFactory(...)

**Reply by shumonsharif** [reply]
@fiodorov70 I didn't test this myself, but I see no reason for it not to work in both JVM and native mode ... please let us know if there is a specific issue you're running into?  In addition to what you're already doing with AbstractPhaseInterceptor, you can also take an approach like the below.

```java
@ApplicationScoped
public class HttpConduitConfiguration {

    private static final Logger LOGGER = Logger.getLogger(HttpConduitConfiguration.class);

    void onStart(@Observes StartupEvent ev) {

        LOGGER.info("Configuring CXF HTTP conduit");

        HTTPConduitConfigurer httpConduitConfigurer = new HTTPConduitConfigurer() {
            public void configure(String name, String address, HTTPConduit c) {
                TLSClientParameters tlsClientParameters = new TLSClientParameters();
                tlsClientParameters.setSSLSocketFactory(...);
                c.setTlsClientParameters(tlsClientParameters);
            }
        };

        final Bus bus = BusFactory.getThreadDefaultBus();
        bus.setExtension(httpConduitConfigurer, HTTPConduitConfigurer.class);
    }
}
```
